<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="Akshay-Udhane" failures="16" tests="20" name="Write Data Service Tests" time="45.819" errors="0" timestamp="2025-10-07T11:02:39 IST">
  <testcase classname="tests.WriteDataServiceTest" name="testConcurrentWriteOperations" time="12.161">
    <failure type="java.lang.AssertionError" message="At least 60% of concurrent write operations should succeed expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: At least 60% of concurrent write operations should succeed expected [true] but found [false]
at tests.WriteDataServiceTest.testConcurrentWriteOperations(WriteDataServiceTest.java:175)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testConcurrentWriteOperations -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteDataServiceHealth" time="0.019">
    <failure type="java.lang.AssertionError" message="Write Data service should be healthy expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Write Data service should be healthy expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteDataServiceHealth(WriteDataServiceTest.java:137)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteDataServiceHealth -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeErrorHandling" time="0.011"/>
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodePerformance" time="0.013"/>
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeValidation" time="0.01">
    <failure type="java.lang.AssertionError" message="Should get valid response for node: ns=3;s=DataBlocksGlobal with value: 42 expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for node: ns=3;s=DataBlocksGlobal with value: 42 expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeValidation(WriteDataServiceTest.java:204)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeValidation -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithBoolean" time="0.01">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 503 expected [200] but found [503]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 503 expected [200] but found [503]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.WriteDataServiceTest.testWriteNodeWithBoolean(WriteDataServiceTest.java:58)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithBoolean -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDifferentDataTypes" time="0.01">
    <failure type="java.lang.AssertionError" message="Should get valid response for Integer value expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for Integer value expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeWithDifferentDataTypes(WriteDataServiceTest.java:84)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDifferentDataTypes -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDifferentDataTypes" time="0.012">
    <failure type="java.lang.AssertionError" message="Should get valid response for String value expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for String value expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeWithDifferentDataTypes(WriteDataServiceTest.java:84)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDifferentDataTypes -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDifferentDataTypes" time="0.012">
    <failure type="java.lang.AssertionError" message="Should get valid response for Boolean value expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for Boolean value expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeWithDifferentDataTypes(WriteDataServiceTest.java:84)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDifferentDataTypes -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDifferentDataTypes" time="0.016">
    <failure type="java.lang.AssertionError" message="Should get valid response for Double value expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for Double value expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeWithDifferentDataTypes(WriteDataServiceTest.java:84)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDifferentDataTypes -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDifferentDataTypes" time="0.011">
    <failure type="java.lang.AssertionError" message="Should get valid response for Numeric node ID with integer expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for Numeric node ID with integer expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeWithDifferentDataTypes(WriteDataServiceTest.java:84)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDifferentDataTypes -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDifferentDataTypes" time="0.009">
    <failure type="java.lang.AssertionError" message="Should get valid response for Another numeric node ID with string expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for Another numeric node ID with string expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeWithDifferentDataTypes(WriteDataServiceTest.java:84)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDifferentDataTypes -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDifferentDataTypes" time="0.009">
    <failure type="java.lang.AssertionError" message="Should get valid response for Invalid node ID expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for Invalid node ID expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeWithDifferentDataTypes(WriteDataServiceTest.java:84)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDifferentDataTypes -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDifferentDataTypes" time="0.009">
    <failure type="java.lang.AssertionError" message="Should get valid response for Empty node ID expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for Empty node ID expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeWithDifferentDataTypes(WriteDataServiceTest.java:84)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDifferentDataTypes -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDifferentDataTypes" time="0.008">
    <failure type="java.lang.AssertionError" message="Should get valid response for Empty string value expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Should get valid response for Empty string value expected [true] but found [false]
at tests.WriteDataServiceTest.testWriteNodeWithDifferentDataTypes(WriteDataServiceTest.java:84)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDifferentDataTypes -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithDouble" time="0.009">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 503 expected [200] but found [503]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 503 expected [200] but found [503]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.WriteDataServiceTest.testWriteNodeWithDouble(WriteDataServiceTest.java:72)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithDouble -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithInteger" time="0.008">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 503 expected [200] but found [503]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 503 expected [200] but found [503]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.WriteDataServiceTest.testWriteNodeWithInteger(WriteDataServiceTest.java:30)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithInteger -->
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithLargeData" time="0.012"/>
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithNullValue" time="0.009"/>
  <testcase classname="tests.WriteDataServiceTest" name="testWriteNodeWithString" time="0.017">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 503 expected [200] but found [503]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 503 expected [200] but found [503]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.WriteDataServiceTest.testWriteNodeWithString(WriteDataServiceTest.java:44)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testWriteNodeWithString -->
</testsuite> <!-- Write Data Service Tests -->
